app/scripts/controllers/preferences.js

```diff
@@ -679,6 +681,31 @@ export default class PreferencesController {
     });
   }
 
+  _subscribeToInfuraAvailability() {
+    this.network.on(NETWORK_EVENTS.INFURA_IS_BLOCKED, () => {
+      this._setInfuraBlocked(true);
+    });
+    this.network.on(NETWORK_EVENTS.INFURA_IS_UNBLOCKED, () => {
+      this._setInfuraBlocked(false);
+    });
+  }
+
+  /**
+   * A setter for the `infuraBlocked` property
+   * @param {boolean} isBlocked - Bool indicating whether Infura is blocked
+   * @returns {Promise<boolean>} A promise of the updated blockage status
+   */
+  _setInfuraBlocked(isBlocked) {
+    const { infuraBlocked } = this.store.getState();
+
+    if (infuraBlocked === isBlocked) {
+      return Promise.resolve(isBlocked);
```

Comment:
Why does this return a Promise if it's synchronous :thinking:

Reply:
Good catch! I had this in line with other setter methods, but given it's usage only in private context, no need to return Promises here